File: Assets/Editor/RemoveFieldFromAnim.cs
using UnityEngine;
using UnityEditor;
using System.Collections.Generic;
using System.IO;

public class RemoveFieldFromAnim : EditorWindow
{
    private string fieldToRemove = "";
    private HashSet<string> inputFilePaths = new HashSet<string>();

    [MenuItem("YuuTools/Animation/AnimPath/Remove Field From Anim")]
    private static void ShowWindow()
    {
        GetWindow<RemoveFieldFromAnim>("Remove Path Field");
    }

    private void OnGUI()
    {
        GUILayout.Label("Field to Remove", EditorStyles.boldLabel);
        fieldToRemove = EditorGUILayout.TextField(fieldToRemove);

        // 检测是否有动画剪辑被拖拽到窗口
        Event evt = Event.current;
        Rect dropArea = GUILayoutUtility.GetRect(0f, 50f, GUILayout.ExpandWidth(true));
        GUI.Box(dropArea, "Drop Animation Clips Here");

        switch (evt.type)
        {
            case EventType.DragUpdated:
            case EventType.DragPerform:
                if (!dropArea.Contains(evt.mousePosition))
                    break;

                DragAndDrop.visualMode = DragAndDropVisualMode.Copy;

                if (evt.type == EventType.DragPerform)
                {
                    DragAndDrop.AcceptDrag();

                    foreach (var draggedObject in DragAndDrop.objectReferences)
                    {
                        if (draggedObject is AnimationClip)
                        {
                            string clipPath = AssetDatabase.GetAssetPath(draggedObject);
                            inputFilePaths.Add(clipPath);
                        }
                    }
                }
                Event.current.Use();
                break;
        }

        if (inputFilePaths.Count > 0)
        {
            GUILayout.Space(10);
            GUILayout.Label("Files to Process:", EditorStyles.boldLabel);
            foreach (string filePath in inputFilePaths)
            {
                GUILayout.Label(filePath);
            }
            GUILayout.Space(10);
        }

        if (inputFilePaths.Count > 0 && GUILayout.Button("Remove Field"))
        {
            RemoveFieldFromAnimFiles();
        }
    }

    private void RemoveFieldFromAnimFiles()
    {
        if (string.IsNullOrEmpty(fieldToRemove) || inputFilePaths.Count == 0)
        {
            Debug.LogError("Field to Remove and Input File Paths cannot be empty!");
            return;
        }

        foreach (string inputFilePath in inputFilePaths)
        {
            // 生成一个临时文件路径
            string tempFilePath = inputFilePath + ".temp";

            using (StreamReader reader = new StreamReader(inputFilePath))
            using (StreamWriter writer = new StreamWriter(tempFilePath))
            {
                string line;
                while ((line = reader.ReadLine()) != null)
                {
                    if (line.StartsWith("    path:"))
                    {
                        // 使用字符串的 Replace 方法删除指定字段
                        string modifiedLine = line.Replace(fieldToRemove, "");
                        writer.WriteLine(modifiedLine);
                    }
                    else
                    {
                        writer.WriteLine(line);
                    }
                }
            }

            // 关闭文件后，将临时文件内容覆盖原文件
            File.Replace(tempFilePath, inputFilePath, null);

            Debug.Log($"处理完成！{inputFilePath}");
        }

        // 清空处理完的文件列表
        inputFilePaths.Clear();
    }
}

File: ScriptOriginVariant/RemoveFieldFromAnim.py
import sys
import os

def remove_field_from_anim_file(input_file_path, field_to_remove):
    # 生成一个临时文件路径
    temp_file_path = input_file_path + ".temp"

    with open(input_file_path, 'r') as input_file, open(temp_file_path, 'w') as temp_file:
        for line in input_file:
            if line.startswith("    path:"):
                # 使用字符串的replace方法删除指定字段
                modified_line = line.replace(field_to_remove, "")
                temp_file.write(modified_line)
            else:
                temp_file.write(line)

    # 关闭文件后，将临时文件内容覆盖原文件
    os.replace(temp_file_path, input_file_path)

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python script.py input_file field_to_remove")
        sys.exit(1)

    input_file_path = sys.argv[1]
    field_to_remove = sys.argv[2]

    remove_field_from_anim_file(input_file_path, field_to_remove)

    print("处理完成！")

